kia-lib/src/main/kotlin/com/faforever/ice/telemetry/ProtocolV1.kt

```diff
@@ -47,15 +49,31 @@ data class UpdateCoturnList(
 
 @JvmRecord
 data class UpdateGameState(
-    val newState: String, // TODO: Use GameState enum
+    val newState: GameState,
     override val messageId: UUID = UUID.randomUUID(),
 ) : OutgoingMessageV1
 
 @JvmRecord
 data class UpdateGpgnetState(
-    val newState: String, // TODO: Use GameState enum
+    val newState: GpgnetState,
     override val messageId: UUID = UUID.randomUUID(),
-) : OutgoingMessageV1
+) : OutgoingMessageV1 {
+    enum class GpgnetState {
+        WAITING_FOR_GAME,
+        GAME_CONNECTED,
+        ;
+        companion object {
+            fun from(state: GpgnetProxy.ConnectionState): GpgnetState {
+                return when (state) {
+                    GpgnetProxy.ConnectionState.LISTENING,
+                    GpgnetProxy.ConnectionState.DISCONNECTED,
+                    -> UpdateGpgnetState.GpgnetState.WAITING_FOR_GAME
```

Comment:
The java adapter only sends `DISCONNECTED` or `WAITING_FOR_GAME`.  [Looks ](https://github.com/FAForever/faf-telemetry-server/blob/ac1f78cdd154844beb0493bcddfa1828de9ae7e3/src/main/resources/public/app.js#L62) like the telemetry server has a third one: `OFFLINE`. I wasn't sure if that was intended to be used for the `DISCONNECTED` state or just a general fallback so I just followed the java implementation.